### DISCUSSION TITLE
Are the possible return values of the API calls documented  somewhere?

### DISCUSSION THREAD

**Discussion by Panda88CO** [timdorr/tesla-api#554] [Q&A]
Hi
I have a need to map possible return values from a returned API parameter into integer values.  E.,g. what are the possible return values that can occur for a car's charging state?  Is it documented somewhere?  If not it would be a great document to have :-)

Thanks

**Comment by SylvainGa** [comment]
I've asked the same question and didn't get an answer. This is what I found out and using. This is based on standard errors with a few modification for Tesla

```
	"400" => "Bad_Request",
	"401" => "Unauthorized",
	"402" => "Payment_Required",
	"403" => "Forbidden",
	"404" => "Not_Found",
	"405" => "Method_Not_Allowed",
	"406" => "Not_Acceptable",
	"407" => "Proxy_Authentication_Required",
	"408" => "Request_Timeout",
	"409" => "Conflict",
	"410" => "Gone",
	"411" => "Length_Required",
	"412" => "Precondition_Failed",
	"413" => "Request_Too_Large",
	"414" => "Request-URI_Too_Long",
	"415" => "Unsupported_Media_Type",
	"416" => "Range_Not_Satisfiable",
	"417" => "Expectation_Failed",
	"500" => "Internal_Server_Error",
	"501" => "Not_Implemented",
	"502" => "Bad_Gateway",
	"503" => "Service_Unavailable",
	"504" => "Gateway_Timeout",
	"505" => "HTTP_Version_Not_Supported",
	"511" => "Network_Authentication_Required",
	"540" => "Vehicle_Server_Error"
```

**Reply by Panda88CO** [reply]
Thanks
That helps
I was actually looking for possible return values of a given parameter returned by the API. I need to translate the returned string to a number to allow it to integrate in a different platform

E.g. from charging state:
"response": {
"battery_heater_on": false,
. . . . . .
"charger_power": 0,
"charger_voltage": 0,
**_"charging_state": "Complete",_**
"conn_charge_cable": "SAE",
"est_battery_range": 171.24,

What other values than **_Complete_** exists??

Is that documented somewhere?

**Comment by GaPhi** [comment]
Here are the values I have already seen for that field.
I have the same for all other fields if someone wants to create the
documentation...

        case "charge_state.charging_state":
            return ![ null, // Unavailable
                      "Disconnected", // Disconnected
                      "NoPower", // Connected, no power
                      "Starting", // Connected, starting charge
                      "Charging", // Connected, charging
                      "Stopped", // Connected, charge stopped
                      "Complete", // Connected, charge complete
                    ].includes(d.after)

**Reply by SylvainGa** [reply]
Oh, sorry. I should have read better :) This was posted already in #207 

https://github.com/timdorr/tesla-api/issues/207

**Reply by Panda88CO** [reply]
If you can forward the values, I can take a look at documenting it

**Comment by Panda88CO** [comment]
thanks

Christian Olgaard
________________________________

**Comment by GaPhi** [comment]
Here is my NodeRed node that is in charge of detecting any change in vehicleData command response.
(It also fixes the response for some values but this is just to simplify the remaining logic of the numerous flows)

[delta.zip](https://github.com/timdorr/tesla-api/files/8386988/delta.zip)
